SampleApp/SampleApp/PushSettingsViewController.xaml.cs

```diff
@@ -0,0 +1,52 @@
+﻿/*
+ Copyright Airship and Contributors
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Plugin.Clipboard;
+
+using Xamarin.Forms;
+
+using UrbanAirship.NETStandard;
+
+namespace SampleApp
+{
+    public partial class PushSettingsViewController : ContentPage
+    {
+        public PushSettingsViewController()
+        {
+            InitializeComponent();
+        }
+
+        protected override void OnAppearing()
+        {
+            base.OnAppearing();
+            enabledPushSwitch.On = Airship.Instance.UserNotificationsEnabled;
+            channelId.Detail = Airship.Instance.ChannelId != null ? Airship.Instance.ChannelId : AppResources.none;
+            namedUser.Placeholder = Airship.Instance.NamedUser != null ? Airship.Instance.NamedUser : AppResources.channel_id_cell_placeholder;
+        }
+
+        void SwitchCell_OnChanged(object sender, EventArgs e)
+        {
+            Airship.Instance.UserNotificationsEnabled = enabledPushSwitch.On;
+        }
+
+        void CopyChannelID(object sender, EventArgs e)
+        {
+            if (Airship.Instance.ChannelId != null)
+            {
```

Comment:
You should also update `channelId.Detail` in case the channel was null when `OnAppearing()` ran, e.g.
```
channelId.Detail = Airship.Instance.ChannelId;
```

Reply:
I'am not sure I understand. I already have this line from the `OnAppearing()`  method`channelId.Detail = Airship.Instance.ChannelId != null ? Airship.Instance.ChannelId : AppResources.none;`

Reply:
If the channel is created _after_ OnAppearing() is called, the ID will not appear in the UI. However, the user can still tap on the placeholder text and copy the actual ID. If they do that, it would be a good opportunity to update the UI with the current channel ID.

Reply:
If we want to handle that edge case we should respond to an event that fires when the channel is created, instead of adding side effects to this method. I think it's fine to save that for later follow-up work.